packages/react-components/react-provider/src/components/FluentProvider/FluentProvider.tsx

```diff
@@ -14,4 +16,17 @@ export const FluentProvider = React.forwardRef<HTMLElement, FluentProviderProps>
   return renderFluentProvider_unstable(state, contextValues);
 });
 
+export const FluentProvider = React.forwardRef<HTMLElement, FluentProviderProps>((props, ref) => {
+  const { dir } = useFluent();
+
+  // "FluentProvider" is a wrapper component that renders to ensure that "TextDirectionProvider" will be applied
+  // Otherwise "useStyles" hook inside "useFluentProviderStyles" will return styles for an incorrect direction
+
+  return (
+    <TextDirectionProvider dir={props.dir || dir}>
```

Comment:
I don't like how the `Inner` component adds complexity here, especially considering re-compose scenario.
It would be great if we could encapsulate this deep into the component.

Have you considered not adding this extra layer and instead updating `useFluentProviderStyles`?
- we can have `left` and `right` setting `textAlign` with `@noflip` slots in `makeStyles` there
- apply `left` or `right` based on `state.dir`

Reply:
btw `makeStyles` typings cry when you use `@noflip` on `textAlign` :-/

Reply:
> I don't like how the Inner component adds complexity here, especially considering re-compose scenario.

I also don't like that option.

Another option would be to make `TextDirectionProvider` (coming from `@fluentui/react-components`) public API instead of `dir` prop on `FluentProvider`. Then the problem is solved, but it's not intuitive at all.

> Have you considered not adding this extra layer and instead updating `useFluentProviderStyles`?

Yes, but then in #22933 I have:

```tsx
  const classes = useStyles();
  const themeClassName = useThemeClassName();

  const className = mergeClasses(themeClassName, classes.root);
```

And as `themeClassName` contains classes from `useFluentProviderStyles()` it logs an error because of direction mismatch in `mergeClasses()`:

https://github.com/microsoft/griffel/blob/007d25262e5f0f9924db915cb531a7071f05bb87/packages/core/src/mergeClasses.ts#L109-L113

Reply:
Refactored to use `@griffel/core` and `makeStyles()` from there.

Reply:
Much cleaner now, thanks!